### ISSUE TITLE
build problem with conan for openssl/1.1.1s

### ISSUE THREAD

**Issue by Jannik746** [openssl/openssl#20512]
I tried building libcurl with conan and i keep getting the troubleshooting that my build fails: 
```
ERROR: openssl/1.1.1s: Error in build() method, line 724
        autotools.make()
        ConanException: Error 2 while executing make -j12
```
The troubleshooting only occurs by building openssl/1.1.1s.
Is there a way around this or a fix?
If more informations would be helpful let me know.

My configdata.pm --dump (later my output and command)
```
Command line (with current working directory = .):

    perl ./Configure mingw-conan-Release-Windows-x86_64-gcc-11.2 no-shared --prefix=/ --openssldir=res no-unit-test threads PERL=perl no-tests --release --libdir=lib no-md2

Perl information:

    perl
    5.36.0 for x86_64-msys-thread-multi

Enabled features:

    aria
    asm
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    camellia
    capieng
    cast
    chacha
    cmac
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    ec
    ec2m
    ecdh
    ecdsa
    engine
    err
    filenames
    gost
    hw(-.+)?
    idea
    makedepend
    md4
    mdc2
    multiblock
    nextprotoneg
    pinshared
    ocb
    ocsp
    pic
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    seed
    siphash
    sm2
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    static-engine
    stdio
    threads
    tls
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    afalgeng                [not-linux] OPENSSL_NO_AFALGENG
    asan                    [default]   OPENSSL_NO_ASAN
    buildtest-c++           [default]
    crypto-mdebug           [default]   OPENSSL_NO_CRYPTO_MDEBUG
    crypto-mdebug-backtrace [default]   OPENSSL_NO_CRYPTO_MDEBUG_BACKTRACE
    devcryptoeng            [default]   OPENSSL_NO_DEVCRYPTOENG
    dynamic-engine          [cascade]
    ec_nistp_64_gcc_128     [default]   OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                     [default]   OPENSSL_NO_EGD
    external-tests          [default]   OPENSSL_NO_EXTERNAL_TESTS
    fuzz-libfuzzer          [default]   OPENSSL_NO_FUZZ_LIBFUZZER
    fuzz-afl                [default]   OPENSSL_NO_FUZZ_AFL
    heartbeats              [default]   OPENSSL_NO_HEARTBEATS
    md2                     [option]    OPENSSL_NO_MD2 (skip crypto/md2)
    msan                    [default]   OPENSSL_NO_MSAN
    rc5                     [default]   OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                    [default]   OPENSSL_NO_SCTP
    shared                  [option]
    ssl-trace               [default]   OPENSSL_NO_SSL_TRACE
    tests                   [option]    OPENSSL_NO_TESTS
    ubsan                   [default]   OPENSSL_NO_UBSAN
    unit-test               [option]    OPENSSL_NO_UNIT_TEST
    weak-ssl-ciphers        [default]   OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                    [default]
    zlib-dynamic            [default]
    ssl3                    [default]   OPENSSL_NO_SSL3
    ssl3-method             [default]   OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "r",
    CC => "gcc",
    CFLAGS => "-Wall -O3",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    aes_asm_src => "aes_core.c aes_cbc.c vpaes-x86_64.s aesni-x86_64.s aesni-sha1-x86_64.s aesni-sha256-x86_64.s aesni-mb-x86_64.s",
    aes_obj => "aes_core.o aes_cbc.o vpaes-x86_64.o aesni-x86_64.o aesni-sha1-x86_64.o aesni-sha256-x86_64.o aesni-mb-x86_64.o",
    apps_aux_src => "win32_init.c",
    apps_init_src => "",
    apps_obj => "win32_init.o",
    bf_asm_src => "bf_enc.c",
    bf_obj => "bf_enc.o",
    bn_asm_src => "asm/x86_64-gcc.c x86_64-mont.s x86_64-mont5.s x86_64-gf2m.s rsaz_exp.c rsaz-x86_64.s rsaz-avx2.s",
    bn_obj => "asm/x86_64-gcc.o x86_64-mont.o x86_64-mont5.o x86_64-gf2m.o rsaz_exp.o rsaz-x86_64.o rsaz-avx2.o",
    bn_ops => "SIXTY_FOUR_BIT EXPORT_VAR_AS_FN",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cast_asm_src => "c_enc.c",
    cast_obj => "c_enc.o",
    cflags => "-m64 -m64 -O3",
    chacha_asm_src => "chacha-x86_64.s",
    chacha_obj => "chacha-x86_64.o",
    cmll_asm_src => "cmll-x86_64.s cmll_misc.c",
    cmll_obj => "cmll-x86_64.o cmll_misc.o",
    cppflags => "-DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT",
    cpuid_asm_src => "x86_64cpuid.s",
    cpuid_obj => "x86_64cpuid.o",
    cxxflags => "-m64 -O3",
    defines => [ "NDEBUG" ],
    des_asm_src => "des_enc.c fcrypt_b.c",
    des_obj => "des_enc.o fcrypt_b.o",
    disable => [  ],
    dso_extension => ".dll",
    dso_scheme => "win32",
    ec_asm_src => "ecp_nistz256.c ecp_nistz256-x86_64.s x25519-x86_64.s",
    ec_obj => "ecp_nistz256.o ecp_nistz256-x86_64.o x25519-x86_64.o",
    enable => [  ],
    ex_libs => "-lws2_32 -lgdi32 -lcrypt32",
    exe_extension => ".exe",
    includes => [  ],
    keccak1600_asm_src => "keccak1600-x86_64.s",
    keccak1600_obj => "keccak1600-x86_64.o",
    lflags => "-m64",
    lib_cflags => "",
    lib_cppflags => "-DL_ENDIAN",
    lib_defines => [  ],
    md5_asm_src => "md5-x86_64.s",
    md5_obj => "md5-x86_64.o",
    modes_asm_src => "ghash-x86_64.s aesni-gcm-x86_64.s",
    modes_obj => "ghash-x86_64.o aesni-gcm-x86_64.o",
    module_cflags => "",
    module_cxxflags => "",
    module_ldflags => "-static-libgcc -shared -Wl,--enable-auto-image-base",
    multilib => "64",
    padlock_asm_src => "e_padlock-x86_64.s",
    padlock_obj => "e_padlock-x86_64.o",
    perlasm_scheme => "mingw64",
    poly1305_asm_src => "poly1305-x86_64.s",
    poly1305_obj => "poly1305-x86_64.o",
    rc4_asm_src => "rc4-x86_64.s rc4-md5-x86_64.s",
    rc4_obj => "rc4-x86_64.o rc4-md5-x86_64.o",
    rc5_asm_src => "rc5_enc.c",
    rc5_obj => "rc5_enc.o",
    rmd160_asm_src => "",
    rmd160_obj => "",
    sha1_asm_src => "sha1-x86_64.s sha256-x86_64.s sha512-x86_64.s sha1-mb-x86_64.s sha256-mb-x86_64.s",
    sha1_obj => "sha1-x86_64.o sha256-x86_64.o sha512-x86_64.o sha1-mb-x86_64.o sha256-mb-x86_64.o",
    shared_cflag => "",
    shared_cppflags => "_WINDLL",
    shared_defflag => "",
    shared_defines => [  ],
    shared_extension => ".dll",
    shared_extension_simple => ".dll",
    shared_impflag => "-Wl,--out-implib=",
    shared_import_extension => ".dll.a",
    shared_ldflag => "-static-libgcc -shared -Wl,--enable-auto-image-base",
    shared_rcflag => "--target=pe-x86-64",
    shared_target => "mingw-shared",
    sys_id => "MINGW64",
    thread_defines => [  ],
    thread_scheme => "winthreads",
    unistd => "<unistd.h>",
    uplink_aux_src => "",
    uplink_obj => "",
    wp_asm_src => "wp-x86_64.s",
    wp_obj => "wp-x86_64.o",

Recorded environment:

    AR =
    ARFLAGS =
    AS =
    ASFLAGS =
    BUILDFILE =
    CC =
    CFLAGS =  -m64 -O3
    CPP =
    CPPDEFINES =
    CPPFLAGS =  -DNDEBUG
    CPPINCLUDES =
    CROSS_COMPILE =
    CXX =
    CXXFLAGS =  -m64 -O3
    HASHBANGPERL =
    LD =
    LDFLAGS =  -m64
    LDLIBS =
    MT =
    MTFLAGS =
    OPENSSL_LOCAL_CONFIG_DIR =
    PERL = perl
    RANLIB =
    RC =
    RCFLAGS =
    RM =
    WINDRES =
    __CNF_CFLAGS =
    __CNF_CPPDEFINES =
    __CNF_CPPFLAGS =
    __CNF_CPPINCLUDES =
    __CNF_CXXFLAGS =
    __CNF_LDFLAGS =
    __CNF_LDLIBS =

Makevars:

    AR              = ar
    ARFLAGS         = r
    CC              = gcc
    CFLAGS          =  -m64 -O3
    CPPDEFINES      =
    CPPFLAGS        =  -DNDEBUG
    CPPINCLUDES     =
    CXXFLAGS        =  -m64 -O3
    HASHBANGPERL    = perl
    LDFLAGS         =  -m64
    LDLIBS          =
    PERL            = perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         =

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile

build file:

    Makefile

build file templates:

    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
    Configurations/common.tmpl
```

my command
```
conan install C:\Users\janni\CLionProjects\curlverzeichnis\conan --build=missing
```
my output by building (if you need the whole output let me know)
```
crypto/dso/dso_win32.c:98:5: error: unknown type name 'HINSTANCE'
   98 |     HINSTANCE h = NULL, *p = NULL;
      |     ^~~~~~~~~
crypto/dso/dso_win32.c:98:19: warning: initialization of 'int' from 'void *' makes integer from pointer without a cast
-Wint-conversion]
   98 |     HINSTANCE h = NULL, *p = NULL;
      |                   ^~~~
crypto/dso/dso_win32.c:106:9: warning: implicit declaration of function 'LoadLibraryA' [-Wimplicit-function-declaration]
  106 |     h = LoadLibraryA(filename);
      |         ^~~~~~~~~~~~
crypto/dso/dso_win32.c:107:11: warning: comparison between pointer and integer
  107 |     if (h == NULL) {
      |           ^~
crypto/dso/dso_win32.c:129:11: warning: comparison between pointer and integer
  129 |     if (h != NULL)
      |           ^~
crypto/dso/dso_win32.c:130:9: warning: implicit declaration of function 'FreeLibrary' [-Wimplicit-function-declaration]
  130 |         FreeLibrary(h);
      |         ^~~~~~~~~~~
crypto/dso/dso_win32.c: In function 'win32_unload':
crypto/dso/dso_win32.c:136:5: error: unknown type name 'HINSTANCE'
  136 |     HINSTANCE *p;
      |     ^~~~~~~~~
crypto/dso/dso_win32.c: In function 'win32_bind_func':
crypto/dso/dso_win32.c:163:5: error: unknown type name 'HINSTANCE'
  163 |     HINSTANCE *ptr;
      |     ^~~~~~~~~
crypto/dso/dso_win32.c:166:9: error: unknown type name 'FARPROC'
  166 |         FARPROC f;
      |         ^~~~~~~
crypto/dso/dso_win32.c:182:13: warning: implicit declaration of function 'GetProcAddress' [-Wimplicit-function-declaration]
  182 |     sym.f = GetProcAddress(*ptr, symname);
      |             ^~~~~~~~~~~~~~
gcc  -I. -Iinclude -m64 -m64 -O3 -m64 -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/res\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DNDEBUG -DNDEBUG -DNDEBUG -MMD -MF crypto/ec/ec_asn1.d.tmp -MT crypto/ec/ec_asn1.o -c -o crypto/ec/ec_asn1.o crypto/ec/ec_asn1.c
crypto/dso/dso_win32.c:188:12: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  188 |     return (DSO_FUNC_TYPE)sym.f;
      |            ^
In file included from crypto/dso/dso_win32.c:490:
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h: At top level:
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:15:3: error: unknown type name 'HANDLE'
   15 |   HANDLE WINAPI CreateToolhelp32Snapshot(DWORD dwFlags,DWORD th32ProcessID);
      |   ^~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:15:17: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'CreateToolhelp32Snapshot'
   15 |   HANDLE WINAPI CreateToolhelp32Snapshot(DWORD dwFlags,DWORD th32ProcessID);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:26:5: error: unknown type name 'SIZE_T'
   26 |     SIZE_T dwSize;
      |     ^~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:27:5: error: unknown type name 'DWORD'
   27 |     DWORD th32ProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:28:5: error: unknown type name 'ULONG_PTR'
   28 |     ULONG_PTR th32HeapID;
      |     ^~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:29:5: error: unknown type name 'DWORD'
   29 |     DWORD dwFlags;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:37:3: error: unknown type name 'WINBOOL'
   37 |   WINBOOL WINAPI Heap32ListFirst(HANDLE hSnapshot,LPHEAPLIST32 lphl);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:37:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Heap32ListFirst'
   37 |   WINBOOL WINAPI Heap32ListFirst(HANDLE hSnapshot,LPHEAPLIST32 lphl);
      |                  ^~~~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:38:3: error: unknown type name 'WINBOOL'
   38 |   WINBOOL WINAPI Heap32ListNext(HANDLE hSnapshot,LPHEAPLIST32 lphl);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:38:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Heap32ListNext'
   38 |   WINBOOL WINAPI Heap32ListNext(HANDLE hSnapshot,LPHEAPLIST32 lphl);
      |                  ^~~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:41:5: error: unknown type name 'SIZE_T'
   41 |     SIZE_T dwSize;
      |     ^~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:42:5: error: unknown type name 'HANDLE'
   42 |     HANDLE hHandle;
      |     ^~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:43:5: error: unknown type name 'ULONG_PTR'
   43 |     ULONG_PTR dwAddress;
      |     ^~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:44:5: error: unknown type name 'SIZE_T'
   44 |     SIZE_T dwBlockSize;
      |     ^~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:45:5: error: unknown type name 'DWORD'
   45 |     DWORD dwFlags;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:46:5: error: unknown type name 'DWORD'
   46 |     DWORD dwLockCount;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:47:5: error: unknown type name 'DWORD'
   47 |     DWORD dwResvd;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:48:5: error: unknown type name 'DWORD'
   48 |     DWORD th32ProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:49:5: error: unknown type name 'ULONG_PTR'
   49 |     ULONG_PTR th32HeapID;
      |     ^~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:58:3: error: unknown type name 'WINBOOL'
   58 |   WINBOOL WINAPI Heap32First(LPHEAPENTRY32 lphe,DWORD th32ProcessID,ULONG_PTR th32HeapID);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:58:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Heap32First'
   58 |   WINBOOL WINAPI Heap32First(LPHEAPENTRY32 lphe,DWORD th32ProcessID,ULONG_PTR th32HeapID);
      |                  ^~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:59:3: error: unknown type name 'WINBOOL'
   59 |   WINBOOL WINAPI Heap32Next(LPHEAPENTRY32 lphe);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:59:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Heap32Next'
   59 |   WINBOOL WINAPI Heap32Next(LPHEAPENTRY32 lphe);
      |                  ^~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:60:3: error: unknown type name 'WINBOOL'
   60 |   WINBOOL WINAPI Toolhelp32ReadProcessMemory(DWORD th32ProcessID,LPCVOID lpBaseAddress,LPVOID lpBuffer,SIZE_T cbRead,SIZE_T *lpNumberOfBytesRead);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:60:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Toolhelp32ReadProcessMemory'
   60 |   WINBOOL WINAPI Toolhelp32ReadProcessMemory(DWORD th32ProcessID,LPCVOID lpBaseAddress,LPVOID lpBuffer,SIZE_T cbRead,SIZE_T *lpNumberOfBytesRead);
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from crypto/dso/dso_win32.c:490:
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:63:5: error: unknown type name 'DWORD'
   63 |     DWORD dwSize;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:64:5: error: unknown type name 'DWORD'
   64 |     DWORD cntUsage;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:65:5: error: unknown type name 'DWORD'
   65 |     DWORD th32ProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:66:5: error: unknown type name 'ULONG_PTR'
   66 |     ULONG_PTR th32DefaultHeapID;
      |     ^~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:67:5: error: unknown type name 'DWORD'
   67 |     DWORD th32ModuleID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:68:5: error: unknown type name 'DWORD'
   68 |     DWORD cntThreads;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:69:5: error: unknown type name 'DWORD'
   69 |     DWORD th32ParentProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:70:5: error: unknown type name 'LONG'
   70 |     LONG pcPriClassBase;
      |     ^~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:71:5: error: unknown type name 'DWORD'
   71 |     DWORD dwFlags;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:72:5: error: unknown type name 'WCHAR'
   72 |     WCHAR szExeFile[MAX_PATH];
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:72:21: error: 'MAX_PATH' undeclared here (not in a function)
   72 |     WCHAR szExeFile[MAX_PATH];
      |                     ^~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:77:3: error: unknown type name 'WINBOOL'
   77 |   WINBOOL WINAPI Process32FirstW(HANDLE hSnapshot,LPPROCESSENTRY32W lppe);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:77:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Process32FirstW'
   77 |   WINBOOL WINAPI Process32FirstW(HANDLE hSnapshot,LPPROCESSENTRY32W lppe);
      |                  ^~~~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:78:3: error: unknown type name 'WINBOOL'
   78 |   WINBOOL WINAPI Process32NextW(HANDLE hSnapshot,LPPROCESSENTRY32W lppe);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:78:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Process32NextW'
   78 |   WINBOOL WINAPI Process32NextW(HANDLE hSnapshot,LPPROCESSENTRY32W lppe);
      |                  ^~~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:81:5: error: unknown type name 'DWORD'
   81 |     DWORD dwSize;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:82:5: error: unknown type name 'DWORD'
   82 |     DWORD cntUsage;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:83:5: error: unknown type name 'DWORD'
   83 |     DWORD th32ProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:84:5: error: unknown type name 'ULONG_PTR'
   84 |     ULONG_PTR th32DefaultHeapID;
      |     ^~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:85:5: error: unknown type name 'DWORD'
   85 |     DWORD th32ModuleID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:86:5: error: unknown type name 'DWORD'
   86 |     DWORD cntThreads;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:87:5: error: unknown type name 'DWORD'
   87 |     DWORD th32ParentProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:88:5: error: unknown type name 'LONG'
   88 |     LONG pcPriClassBase;
      |     ^~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:89:5: error: unknown type name 'DWORD'
   89 |     DWORD dwFlags;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:90:5: error: unknown type name 'CHAR'
   90 |     CHAR szExeFile[MAX_PATH];
      |     ^~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:95:3: error: unknown type name 'WINBOOL'
   95 |   WINBOOL WINAPI Process32First(HANDLE hSnapshot,LPPROCESSENTRY32 lppe);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:95:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Process32First'
   95 |   WINBOOL WINAPI Process32First(HANDLE hSnapshot,LPPROCESSENTRY32 lppe);
      |                  ^~~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:96:3: error: unknown type name 'WINBOOL'
   96 |   WINBOOL WINAPI Process32Next(HANDLE hSnapshot,LPPROCESSENTRY32 lppe);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:96:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Process32Next'
   96 |   WINBOOL WINAPI Process32Next(HANDLE hSnapshot,LPPROCESSENTRY32 lppe);
      |                  ^~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:107:5: error: unknown type name 'DWORD'
  107 |     DWORD dwSize;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:108:5: error: unknown type name 'DWORD'
  108 |     DWORD cntUsage;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:109:5: error: unknown type name 'DWORD'
  109 |     DWORD th32ThreadID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:110:5: error: unknown type name 'DWORD'
  110 |     DWORD th32OwnerProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:111:5: error: unknown type name 'LONG'
  111 |     LONG tpBasePri;
      |     ^~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:112:5: error: unknown type name 'LONG'
  112 |     LONG tpDeltaPri;
      |     ^~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:113:5: error: unknown type name 'DWORD'
  113 |     DWORD dwFlags;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:118:3: error: unknown type name 'WINBOOL'
  118 |   WINBOOL WINAPI Thread32First(HANDLE hSnapshot,LPTHREADENTRY32 lpte);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:118:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Thread32First'
  118 |   WINBOOL WINAPI Thread32First(HANDLE hSnapshot,LPTHREADENTRY32 lpte);
      |                  ^~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:119:3: error: unknown type name 'WINBOOL'
  119 |   WINBOOL WINAPI Thread32Next(HANDLE hSnapshot,LPTHREADENTRY32 lpte);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:119:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Thread32Next'
  119 |   WINBOOL WINAPI Thread32Next(HANDLE hSnapshot,LPTHREADENTRY32 lpte);
      |                  ^~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:122:5: error: unknown type name 'DWORD'
  122 |     DWORD dwSize;
      |     ^~~~~
gcc  -I. -Iinclude -m64 -m64 -O3 -m64 -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/res\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DNDEBUG -DNDEBUG -DNDEBUG -MMD -MF crypto/ec/ec_check.d.tmp -MT crypto/ec/ec_check.o -c -o crypto/ec/ec_check.o crypto/ec/ec_check.c
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:123:5: error: unknown type name 'DWORD'
  123 |     DWORD th32ModuleID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:124:5: error: unknown type name 'DWORD'
  124 |     DWORD th32ProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:125:5: error: unknown type name 'DWORD'
  125 |     DWORD GlblcntUsage;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:126:5: error: unknown type name 'DWORD'
  126 |     DWORD ProccntUsage;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:127:5: error: unknown type name 'BYTE'
  127 |     BYTE *modBaseAddr;
      |     ^~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:128:5: error: unknown type name 'DWORD'
  128 |     DWORD modBaseSize;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:129:5: error: unknown type name 'HMODULE'
  129 |     HMODULE hModule;
      |     ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:130:5: error: unknown type name 'WCHAR'
  130 |     WCHAR szModule[MAX_MODULE_NAME32 + 1];
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:131:5: error: unknown type name 'WCHAR'
  131 |     WCHAR szExePath[MAX_PATH];
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:136:3: error: unknown type name 'WINBOOL'
  136 |   WINBOOL WINAPI Module32FirstW(HANDLE hSnapshot,LPMODULEENTRY32W lpme);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:136:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Module32FirstW'
  136 |   WINBOOL WINAPI Module32FirstW(HANDLE hSnapshot,LPMODULEENTRY32W lpme);
      |                  ^~~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:137:3: error: unknown type name 'WINBOOL'
  137 |   WINBOOL WINAPI Module32NextW(HANDLE hSnapshot,LPMODULEENTRY32W lpme);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:137:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Module32NextW'
  137 |   WINBOOL WINAPI Module32NextW(HANDLE hSnapshot,LPMODULEENTRY32W lpme);
      |                  ^~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:140:5: error: unknown type name 'DWORD'
  140 |     DWORD dwSize;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:141:5: error: unknown type name 'DWORD'
  141 |     DWORD th32ModuleID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:142:5: error: unknown type name 'DWORD'
  142 |     DWORD th32ProcessID;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:143:5: error: unknown type name 'DWORD'
  143 |     DWORD GlblcntUsage;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:144:5: error: unknown type name 'DWORD'
  144 |     DWORD ProccntUsage;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:145:5: error: unknown type name 'BYTE'
  145 |     BYTE *modBaseAddr;
      |     ^~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:146:5: error: unknown type name 'DWORD'
  146 |     DWORD modBaseSize;
      |     ^~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:147:5: error: unknown type name 'HMODULE'
  147 |     HMODULE hModule;
      |     ^~~~~~~
gcc  -I. -Iinclude -m64 -m64 -O3 -m64 -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/res\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DNDEBUG -DNDEBUG -DNDEBUG -MMD -MF crypto/ec/ec_curve.d.tmp -MT crypto/ec/ec_curve.o -c -o crypto/ec/ec_curve.o crypto/ec/ec_curve.c
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:154:3: error: unknown type name 'WINBOOL'
  154 |   WINBOOL WINAPI Module32First(HANDLE hSnapshot,LPMODULEENTRY32 lpme);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:154:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Module32First'
  154 |   WINBOOL WINAPI Module32First(HANDLE hSnapshot,LPMODULEENTRY32 lpme);
      |                  ^~~~~~~~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:155:3: error: unknown type name 'WINBOOL'
  155 |   WINBOOL WINAPI Module32Next(HANDLE hSnapshot,LPMODULEENTRY32 lpme);
      |   ^~~~~~~
/c/.conan/a6acd3b/1/bin/msys64/usr/include/w32api/tlhelp32.h:155:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'Module32Next'
  155 |   WINBOOL WINAPI Module32Next(HANDLE hSnapshot,LPMODULEENTRY32 lpme);
      |                  ^~~~~~~~~~~~
crypto/dso/dso_win32.c:500:16: error: unknown type name 'WINAPI'
  500 | typedef HANDLE(WINAPI *CREATETOOLHELP32SNAPSHOT) (DWORD, DWORD);
      |                ^~~~~~
crypto/dso/dso_win32.c:501:14: error: unknown type name 'WINAPI'
  501 | typedef BOOL(WINAPI *CLOSETOOLHELP32SNAPSHOT) (HANDLE);
      |              ^~~~~~
gcc  -I. -Iinclude -m64 -m64 -O3 -m64 -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/res\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DNDEBUG -DNDEBUG -DNDEBUG -MMD -MF crypto/ec/ec_cvt.d.tmp -MT crypto/ec/ec_cvt.o -c -o crypto/ec/ec_cvt.o crypto/ec/ec_cvt.c
crypto/dso/dso_win32.c:502:14: error: unknown type name 'WINAPI'
  502 | typedef BOOL(WINAPI *MODULE32) (HANDLE, MODULEENTRY32 *);
      |              ^~~~~~
crypto/dso/dso_win32.c: In function 'win32_pathbyaddr':
crypto/dso/dso_win32.c:506:5: error: unknown type name 'HMODULE'
  506 |     HMODULE dll;
      |     ^~~~~~~
crypto/dso/dso_win32.c:507:5: error: unknown type name 'HANDLE'
  507 |     HANDLE hModuleSnap = INVALID_HANDLE_VALUE;
      |     ^~~~~~
crypto/dso/dso_win32.c:507:26: error: 'INVALID_HANDLE_VALUE' undeclared (first use in this function)
  507 |     HANDLE hModuleSnap = INVALID_HANDLE_VALUE;
      |                          ^~~~~~~~~~~~~~~~~~~~
gcc  -I. -Iinclude -m64 -m64 -O3 -m64 -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/res\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DNDEBUG -DNDEBUG -DNDEBUG -MMD -MF crypto/ec/ec_err.d.tmp -MT crypto/ec/ec_err.o -c -o crypto/ec/ec_err.o crypto/ec/ec_err.c
crypto/dso/dso_win32.c:507:26: note: each undeclared identifier is reported only once for each function it appears in
crypto/dso/dso_win32.c:509:5: error: unknown type name 'CREATETOOLHELP32SNAPSHOT'
  509 |     CREATETOOLHELP32SNAPSHOT create_snap;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:510:5: error: unknown type name 'CLOSETOOLHELP32SNAPSHOT'
  510 |     CLOSETOOLHELP32SNAPSHOT close_snap;
      |     ^~~~~~~~~~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:511:5: error: unknown type name 'MODULE32'; did you mean 'MODULEENTRY32'?
  511 |     MODULE32 module_first, module_next;
      |     ^~~~~~~~
      |     MODULEENTRY32
crypto/dso/dso_win32.c:523:11: warning: implicit declaration of function 'LoadLibrary' [-Wimplicit-function-declaration]
  523 |     dll = LoadLibrary(TEXT(DLLNAME));
      |           ^~~~~~~~~~~
crypto/dso/dso_win32.c:523:23: warning: implicit declaration of function 'TEXT' [-Wimplicit-function-declaration]
  523 |     dll = LoadLibrary(TEXT(DLLNAME));
      |                       ^~~~
crypto/dso/dso_win32.c:524:13: warning: comparison between pointer and integer
  524 |     if (dll == NULL) {
      |             ^~
crypto/dso/dso_win32.c:529:20: error: 'CREATETOOLHELP32SNAPSHOT' undeclared (first use in this function)
  529 |     create_snap = (CREATETOOLHELP32SNAPSHOT)
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~
gcc  -I. -Iinclude -m64 -m64 -O3 -m64 -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/res\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DNDEBUG -DNDEBUG -DNDEBUG -MMD -MF crypto/ec/ec_key.d.tmp -MT crypto/ec/ec_key.o -c -o crypto/ec/ec_key.o crypto/ec/ec_key.c
crypto/dso/dso_win32.c:529:45: error: expected ';' before 'GetProcAddress'
  529 |     create_snap = (CREATETOOLHELP32SNAPSHOT)
      |                                             ^
      |                                             ;
  530 |         GetProcAddress(dll, "CreateToolhelp32Snapshot");
      |         ~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:531:21: warning: comparison between pointer and integer
  531 |     if (create_snap == NULL) {
      |                     ^~
crypto/dso/dso_win32.c:541:19: error: 'CLOSETOOLHELP32SNAPSHOT' undeclared (first use in this function)
  541 |     close_snap = (CLOSETOOLHELP32SNAPSHOT) CloseHandle;
      |                   ^~~~~~~~~~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:541:43: error: expected ';' before 'CloseHandle'
  541 |     close_snap = (CLOSETOOLHELP32SNAPSHOT) CloseHandle;
      |                                           ^~~~~~~~~~~~
      |                                           ;
crypto/dso/dso_win32.c:543:21: error: 'MODULE32' undeclared (first use in this function)
  543 |     module_first = (MODULE32) GetProcAddress(dll, "Module32First");
      |                     ^~~~~~~~
crypto/dso/dso_win32.c:543:30: error: expected ';' before 'GetProcAddress'
  543 |     module_first = (MODULE32) GetProcAddress(dll, "Module32First");
      |                              ^~~~~~~~~~~~~~~
      |                              ;
crypto/dso/dso_win32.c:544:29: error: expected ';' before 'GetProcAddress'
  544 |     module_next = (MODULE32) GetProcAddress(dll, "Module32Next");
      |                             ^~~~~~~~~~~~~~~
      |                             ;
crypto/dso/dso_win32.c:550:20: error: invalid type argument of unary '*' (have 'int')
  550 |     hModuleSnap = (*create_snap) (TH32CS_SNAPMODULE, 0);
      |                    ^~~~~~~~~~~~
crypto/dso/dso_win32.c:559:11: error: invalid type argument of unary '*' (have 'int')
  559 |     if (!(*module_first) (hModuleSnap, &me32)) {
      |           ^~~~~~~~~~~~~
crypto/dso/dso_win32.c:560:10: error: invalid type argument of unary '*' (have 'int')
  560 |         (*close_snap) (hModuleSnap);
      |          ^~~~~~~~~~~
crypto/dso/dso_win32.c:570:14: error: invalid type argument of unary '*' (have 'int')
  570 |             (*close_snap) (hModuleSnap);
      |              ^~~~~~~~~~~
crypto/dso/dso_win32.c:602:15: error: invalid type argument of unary '*' (have 'int')
  602 |     } while ((*module_next) (hModuleSnap, &me32));
      |               ^~~~~~~~~~~~
crypto/dso/dso_win32.c:604:6: error: invalid type argument of unary '*' (have 'int')
  604 |     (*close_snap) (hModuleSnap);
      |      ^~~~~~~~~~~
crypto/dso/dso_win32.c: In function 'win32_globallookup':
crypto/dso/dso_win32.c:611:5: error: unknown type name 'HMODULE'
  611 |     HMODULE dll;
      |     ^~~~~~~
crypto/dso/dso_win32.c:612:5: error: unknown type name 'HANDLE'
  612 |     HANDLE hModuleSnap = INVALID_HANDLE_VALUE;
      |     ^~~~~~
gcc  -I. -Iinclude -m64 -m64 -O3 -m64 -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/res\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DNDEBUG -DNDEBUG -DNDEBUG -MMD -MF crypto/ec/ec_kmeth.d.tmp -MT crypto/ec/ec_kmeth.o -c -o crypto/ec/ec_kmeth.o crypto/ec/ec_kmeth.c
crypto/dso/dso_win32.c:612:26: error: 'INVALID_HANDLE_VALUE' undeclared (first use in this function)
  612 |     HANDLE hModuleSnap = INVALID_HANDLE_VALUE;
      |                          ^~~~~~~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:614:5: error: unknown type name 'CREATETOOLHELP32SNAPSHOT'
  614 |     CREATETOOLHELP32SNAPSHOT create_snap;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:615:5: error: unknown type name 'CLOSETOOLHELP32SNAPSHOT'
  615 |     CLOSETOOLHELP32SNAPSHOT close_snap;
      |     ^~~~~~~~~~~~~~~~~~~~~~~
gcc  -I. -Iinclude -m64 -m64 -O3 -m64 -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/res\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DNDEBUG -DNDEBUG -DNDEBUG -MMD -MF crypto/ec/ec_lib.d.tmp -MT crypto/ec/ec_lib.o -c -o crypto/ec/ec_lib.o crypto/ec/ec_lib.c
crypto/dso/dso_win32.c:616:5: error: unknown type name 'MODULE32'; did you mean 'MODULEENTRY32'?
  616 |     MODULE32 module_first, module_next;
      |     ^~~~~~~~
      |     MODULEENTRY32
crypto/dso/dso_win32.c:619:9: error: unknown type name 'FARPROC'
  619 |         FARPROC f;
      |         ^~~~~~~
crypto/dso/dso_win32.c:623:13: warning: comparison between pointer and integer
  623 |     if (dll == NULL) {
      |             ^~
crypto/dso/dso_win32.c:628:20: error: 'CREATETOOLHELP32SNAPSHOT' undeclared (first use in this function)
  628 |     create_snap = (CREATETOOLHELP32SNAPSHOT)
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:628:45: error: expected ';' before 'GetProcAddress'
  628 |     create_snap = (CREATETOOLHELP32SNAPSHOT)
      |                                             ^
      |                                             ;
  629 |         GetProcAddress(dll, "CreateToolhelp32Snapshot");
      |         ~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:630:21: warning: comparison between pointer and integer
  630 |     if (create_snap == NULL) {
      |                     ^~
crypto/dso/dso_win32.c:640:19: error: 'CLOSETOOLHELP32SNAPSHOT' undeclared (first use in this function)
  640 |     close_snap = (CLOSETOOLHELP32SNAPSHOT) CloseHandle;
      |                   ^~~~~~~~~~~~~~~~~~~~~~~
crypto/dso/dso_win32.c:640:43: error: expected ';' before 'CloseHandle'
  640 |     close_snap = (CLOSETOOLHELP32SNAPSHOT) CloseHandle;
      |                                           ^~~~~~~~~~~~
      |                                           ;
crypto/dso/dso_win32.c:642:21: error: 'MODULE32' undeclared (first use in this function)
  642 |     module_first = (MODULE32) GetProcAddress(dll, "Module32First");
      |                     ^~~~~~~~
crypto/dso/dso_win32.c:642:30: error: expected ';' before 'GetProcAddress'
  642 |     module_first = (MODULE32) GetProcAddress(dll, "Module32First");
      |                              ^~~~~~~~~~~~~~~
      |                              ;
crypto/dso/dso_win32.c:643:29: error: expected ';' before 'GetProcAddress'
  643 |     module_next = (MODULE32) GetProcAddress(dll, "Module32Next");
      |                             ^~~~~~~~~~~~~~~
      |                             ;
crypto/dso/dso_win32.c:645:20: error: invalid type argument of unary '*' (have 'int')
  645 |     hModuleSnap = (*create_snap) (TH32CS_SNAPMODULE, 0);
      |                    ^~~~~~~~~~~~
crypto/dso/dso_win32.c:654:11: error: invalid type argument of unary '*' (have 'int')
  654 |     if (!(*module_first) (hModuleSnap, &me32)) {
      |           ^~~~~~~~~~~~~
crypto/dso/dso_win32.c:655:10: error: invalid type argument of unary '*' (have 'int')
  655 |         (*close_snap) (hModuleSnap);
      |          ^~~~~~~~~~~
crypto/dso/dso_win32.c:662:14: error: invalid type argument of unary '*' (have 'int')
  662 |             (*close_snap) (hModuleSnap);
      |              ^~~~~~~~~~~
crypto/dso/dso_win32.c:666:15: error: invalid type argument of unary '*' (have 'int')
  666 |     } while ((*module_next) (hModuleSnap, &me32));
      |               ^~~~~~~~~~~~
crypto/dso/dso_win32.c:668:6: error: invalid type argument of unary '*' (have 'int')
  668 |     (*close_snap) (hModuleSnap);
      |      ^~~~~~~~~~~
make[1]: *** [Makefile:2794: crypto/dso/dso_win32.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/c/Users/janni/.conan/data/openssl/1.1.1s/_/_/build/d846067d59c3b90a9880f5caaf4b040d4f30bf14/src'
make: *** [Makefile:188: all] Fehler 2
openssl/1.1.1s:
openssl/1.1.1s: ERROR: Package 'd846067d59c3b90a9880f5caaf4b040d4f30bf14' build failed
openssl/1.1.1s: WARN: Build folder C:\Users\janni\.conan\data\openssl\1.1.1s\_\_\build\d846067d59c3b90a9880f5caaf4b040d4f30bf14\build-release
ERROR: openssl/1.1.1s: Error in build() method, line 724
        autotools.make()
        ConanException: Error 2 while executing make -j12
```

**Comment by paulidale** [comment]
Something is messed up.  OpenSSL doesn't support using conan (or any other third party build system).
Can you configure and build OpenSSL following the instructions in the distribution?
If so, the problem lies with conan I suspect.

**Comment by t8m** [comment]
The version 1.1.1 is not supported anymore. Closing. If this is still an issue with currently supported releases, please open a new issue.
